hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerStateManager.java

```diff
@@ -237,39 +239,33 @@ void loadContainer(final ContainerInfo containerInfo) throws SCMException {
 
   /**
    * Allocates a new container based on the type, replication etc.
-   *
-   * @param pipelineManager -- Pipeline Manager class.
-   * @param type -- Replication type.
-   * @param replicationFactor - Replication replicationFactor.
-   * @return ContainerWithPipeline
-   * @throws IOException  on Failure.
    */
   ContainerInfo allocateContainer(final PipelineManager pipelineManager,
-      final HddsProtos.ReplicationType type,
-      final HddsProtos.ReplicationFactor replicationFactor, final String owner)
+      final ReplicationConfig replicationConfig, final String owner)
       throws IOException {
     final List<Pipeline> pipelines = pipelineManager
-        .getPipelines(type, replicationFactor, Pipeline.PipelineState.OPEN);
+        .getPipelines(replicationConfig, Pipeline.PipelineState.OPEN);
     Pipeline pipeline;
 
-    boolean bgCreateOne = (type == ReplicationType.RATIS) && replicationFactor
-        == ReplicationFactor.ONE && autoCreateRatisOne;
-    boolean bgCreateThree = (type == ReplicationType.RATIS) && replicationFactor
-        == ReplicationFactor.THREE;
+    boolean bgCreateOne = RatisReplicationConfig
+        .hasFactor(replicationConfig, ReplicationFactor.ONE)
+        && autoCreateRatisOne;
+    boolean bgCreateThree = RatisReplicationConfig
+        .hasFactor(replicationConfig, ReplicationFactor.ONE);
```

Comment:
Is this last line of the change a mistake? The original code had THREE but this last line has ONE.

Reply:
yep. Thanks, fixing.